tests/test_smoke.py

```diff
@@ -0,0 +1,81 @@
+import yaml
+import subprocess
+import time
+import pytest
+
+
+def test_install():
+    with open("snap/snapcraft.yaml") as file:
+        snapcraft = yaml.safe_load(file)
+
+        subprocess.run(
+            f"sudo snap install ./{snapcraft['name']}_{snapcraft['version']}_amd64.snap --devmode".split(),
+            check=True,
+        )
+
+
+@pytest.mark.run(after="test_install")
+def test_all_apps():
+    with open("snap/snapcraft.yaml") as file:
+        snapcraft = yaml.safe_load(file)
+
+        override = {
+            "pgbackrest": "version",
+        }
+
+        skip = [
+            "pg-buildext",
+            "pg-conftool",
+            "pg-createcluster",
+            "pg-dropcluster",
+            "pg-receivexlog",
+            "pg-upgradecluster",
+            "pg-backupcluster",
+            "pg-restorecluster",
+            "pg-virtualenv",
+            "pg-ctlcluster",
+            "pg-renamecluster",
+            "pg-updatedicts",
+            "pg-lsclusters",
+        ]
+
+        for app, data in snapcraft["apps"].items():
+            if not bool(data.get("daemon")) and app not in skip:
+                subprocess.run(
+                    f"{snapcraft['name']}.{app} {override.get(app, '--help')}".split(),
+                    check=True,
+                )
+
+
+@pytest.mark.run(after="test_install")
+def test_all_services():
+    with open("snap/snapcraft.yaml") as file:
+        snapcraft = yaml.safe_load(file)
+
+        skip = [
+            "pgbackrest-service",
```

Comment:
There are no pgbackrest in the snap.

BTW, those warnings looks suspicious: https://github.com/canonical/postgresql-snap/actions/runs/13588829263/job/37990079310#step:6:690

Reply:
Registered the mark to remove the warning.